src/modules/deal-guarantee/dto/create-deal-guarantee-request.dto.ts

```diff
@@ -0,0 +1,98 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { PROPERTIES } from '@ukef/constants';
+import { DateOnlyString } from '@ukef/helpers/date-only-string.type';
+import { IsISO8601, IsNotEmpty, IsOptional, Length, Matches, Min, MinLength } from 'class-validator';
+
+export type CreateDealGuaranteeRequest = CreateDealGuaranteeRequestItem[];
+
+export class CreateDealGuaranteeRequestItem {
+  @ApiProperty({
+    description: 'The identifier of the deal to create the guarantee for.',
+    example: '00000001',
+    minLength: 1,
+    maxLength: 10,
+  })
+  @Length(1, 10)
+  // TODO APIM-73: Should we remove dealIdentifier from the request body?
```

Comment:
Now that we have `dealIdentifier` in the URL, should we remove it from the request body?

Reply:
Can client still send it and it would be ignored? or extra fields will cause errors?

If we remove `dealIdentifier` here then it will be more similar changes.

Reply:
Extra fields don't cause validation errors in the code at the moment. In this PR, the controller removes any extra keys without error. So the client can still send it and it would be ignored